mmengine/visualization/visualizer.py

```diff
@@ -215,19 +215,24 @@ def __init__(
         if image is not None:
             self.set_image(image)
 
+    def _should_do(self):
+        if self.master_only and get_rank() != 0:
+            return False
+        else:
+            return True
+
     @property  # type: ignore
     @master_only
     def dataset_meta(self) -> Optional[dict]:
         """Optional[dict]: Meta info of the dataset."""
         return self._dataset_meta
 
     @dataset_meta.setter  # type: ignore
-    @master_only
     def dataset_meta(self, dataset_meta: dict) -> None:
         """Set the dataset meta info to the Visualizer."""
-        self._dataset_meta = dataset_meta
+        if self._should_do():
```

Comment:
We need to reconsider that if it is necessary to apply `self._should_do` here. I think we could skip this check here.

Reply:
For this, I added the check based on the original master_only decorator for this function.  For dataset_meta, it might be better to keep the original master_only decorator.